### DISCUSSION TITLE
Static Media Failure afer move of netbox location

### DISCUSSION THREAD

**Discussion by misch42** [netbox-community/netbox#14757] [Help Wanted!] [answered]
Hi,

I wanted my netbox to run unter /netbox/ location. So I change BASE_PATH in configuration.py to netbox/. I change my httpd.conf to

    Alias /netbox/static/ /opt/netbox/netbox/static

    <Directory /opt/netbox/netbox/static>
        Options Indexes FollowSymLinks MultiViews
        AllowOverride None
        Require all granted
    </Directory>

    <Location /netbox/static>
        ProxyPass !
    </Location>

I re-ran manage.py collectstatic.

Still I get:
The HTTP service (e.g. nginx or Apache) is configured to serve files from the STATIC_ROOT path. Refer to [the installation documentation](https://docs.netbox.dev/en/stable/installation/) for further guidance.

    STATIC_ROOT: /opt/netbox/netbox/static
    STATIC_URL: /netbox/static/

The file netbox-print.css exists in the static root directory and is readable by the HTTP server.

The log looks like

"GET /netbox/ HTTP/1.1" 200 105389 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
"GET /netbox/static/netbox-light.css?v=3.6.6 HTTP/1.1" 404 88343 "https://example.com/netbox/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
"GET /netbox/static/netbox-dark.css?v=3.6.6 HTTP/1.1" 404 88342 "https://example.com/netbox/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
"GET /netbox/static/netbox.js?v=3.6.6 HTTP/1.1" 404 88336 "https://example.com/netbox/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"
"GET /netbox/media-failure/?filename=netbox-print.css HTTP/1.1" 200 1751 "https://example.com/netbox/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:121.0) Gecko/20100101 Firefox/121.0"

Any ideas?

**Comment by candlerb** [comment, marked as answer]
Netbox version? (EDIT: from the logs I guess it's v3.6.6)

> GET /netbox/static/netbox-light.css?v=3.6.6 HTTP/1.1" 404

That's the problem there (404 response), so if you've checked that the file `/opt/netbox/netbox/static/netbox-light.css` *does* exist, then clearly it's your Apache config that's at fault. Did you reload/restart Apache after changing the config?

Also, you haven't shown your whole Apache config, so maybe something else is wrong there.

A couple of years ago I used to run Netbox in this way. According to my records, I used

```
Alias /netbox/static /opt/netbox/netbox/static

<Directory /opt/netbox/netbox/static>
    Options Indexes FollowSymLinks MultiViews
    AllowOverride None
    Require all granted
</Directory>

<Location /netbox/static>
    ProxyPass !
</Location>

ProxyPass /netbox http://127.0.0.1:8001/netbox
ProxyPassReverse /netbox http://127.0.0.1:8001/netbox
```

and `BASE_PATH = 'netbox/'`. This was with Netbox 3.3.9.

There was a bug with 3.3.6 and 3.3.7 which prevented it working: #10837. I don't think that's the same problem as you're seeing, but I've not tested it with any newer versions either.

**Comment by misch42** [comment]
The solution was to put the

 <Location /netbox/static>
  ProxyPass !
</Location>

after my <Location /netbox/static> with the proxy statements.

Thanks for the help.
